<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [rabbitmq-discuss] Not able to get ssl working with rabbitMQ	server
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20Not%20able%20to%20get%20ssl%20working%20with%20rabbitMQ%0A%09server&In-Reply-To=%3C1390902900.87968.YahooMailNeo%40web141406.mail.bf1.yahoo.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="033358.html">
   <LINK REL="Next"  HREF="033254.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[rabbitmq-discuss] Not able to get ssl working with rabbitMQ	server</H1>
    <B>Kausik Chattopadhyay</B> 
    <A HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20Not%20able%20to%20get%20ssl%20working%20with%20rabbitMQ%0A%09server&In-Reply-To=%3C1390902900.87968.YahooMailNeo%40web141406.mail.bf1.yahoo.com%3E"
       TITLE="[rabbitmq-discuss] Not able to get ssl working with rabbitMQ	server">connect_kausik at yahoo.com
       </A><BR>
    <I>Tue Jan 28 09:55:00 GMT 2014</I>
    <P><UL>
        <LI>Previous message: <A HREF="033358.html">[rabbitmq-discuss] Not able to get ssl working with rabbitMQ server
</A></li>
        <LI>Next message: <A HREF="033254.html">[rabbitmq-discuss] RabbitMQ hangs, does not accept connections
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#33376">[ date ]</a>
              <a href="thread.html#33376">[ thread ]</a>
              <a href="subject.html#33376">[ subject ]</a>
              <a href="author.html#33376">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Thank you Gavin.
&#160;Actually I tried with your code-sample.py. It worked and then it worked with my client as well. A bit surprised. This s my ssl_options
&#160;ssl_options = {&quot;ca_certs&quot;: &quot;cacert.pem&quot;,
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;&quot;certfile&quot;: &quot;cert.pem&quot;,
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;&quot;keyfile&quot;: &quot;key.pem&quot;,
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;&quot;cert_reqs&quot;: True}

In the serverside the configuration is as follows:

<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbitmq at ubuntuvm</A>:~$ rabbitmqctl environment

Application environment of node <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbit at ubuntuvm</A> ...
[{auth_backends,[rabbit_auth_backend_internal]},
&#160;{auth_mechanisms,['PLAIN','AMQPLAIN']},
&#160;{backing_queue_module,rabbit_variable_queue},
&#160;{cluster_nodes,{[],disc}},
&#160;{collect_statistics,none},
&#160;{collect_statistics_interval,5000},
&#160;{default_permissions,[&lt;&lt;&quot;.*&quot;&gt;&gt;,&lt;&lt;&quot;.*&quot;&gt;&gt;,&lt;&lt;&quot;.*&quot;&gt;&gt;]},
&#160;{default_user,&lt;&lt;&quot;guest&quot;&gt;&gt;},
&#160;{default_user_tags,[administrator]},
&#160;{default_vhost,&lt;&lt;&quot;/&quot;&gt;&gt;},
&#160;{delegate_count,16},
&#160;{disk_free_limit,1000000000},
&#160;{enabled_plugins_file,&quot;/etc/rabbitmq/enabled_plugins&quot;},
&#160;{error_logger,{file,&quot;/var/log/rabbitmq/<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbit at ubuntuvm.log</A>&quot;}},
&#160;{frame_max,131072},
&#160;{heartbeat,600},
&#160;{hipe_compile,false},
&#160;{included_applications,[]},
&#160;{log_levels,[{connection,info}]},
&#160;{msg_store_file_size_limit,16777216},
&#160;{msg_store_index_module,rabbit_msg_store_ets_index},
&#160;{plugins_dir,&quot;/usr/lib/rabbitmq/lib/rabbitmq_server-3.0.2/sbin/../plugins&quot;},
&#160;{plugins_expand_dir,&quot;/var/lib/rabbitmq/mnesia/<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbit at ubuntuvm-plugins-expand</A>&quot;},
&#160;{queue_index_max_journal_entries,262144},
&#160;{reverse_dns_lookups,false},
&#160;{sasl_error_logger,{file,&quot;/var/log/rabbitmq/<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbit at ubuntuvm-sasl.log</A>&quot;}},
&#160;{server_properties,[]},
&#160;{ssl_cert_login_from,distinguished_name},
&#160;{ssl_listeners,[5671]},
&#160;{ssl_options,[{cacertfile,&quot;/home/rabbitmq/testca/cacert.pem&quot;},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;{certfile,&quot;/home/rabbitmq/server/cert.pem&quot;},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;{keyfile,&quot;/home/rabbitmq/server/key.pem&quot;},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;{verify,verify_peer},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;{fail_if_no_peer_cert,false}]},
&#160;{tcp_listen_options,[binary,
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; {packet,raw},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; {reuseaddr,true},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; {backlog,128},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; {nodelay,true},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; {linger,{true,0}},
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; {exit_on_close,false}]},
&#160;{tcp_listeners,[5672]},
&#160;{trace_vhosts,[]},
&#160;{vm_memory_high_watermark,0.4}]
...done.


Trying to debug why it was failing earlier.




On Monday, January 27, 2014 10:12 PM, Gavin M. Roy &lt;<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">gavinmroy at gmail.com</A>&gt; wrote:
 
Here&#8217;s a gist of both my rabbitmq.config, an example python script using BlockingConnection on 0.9.13 and the interactive session running the script, all working with SSL using pika:

<A HREF="https://gist.github.com/gmr/8651586">https://gist.github.com/gmr/8651586</A>

I&#8217;m going to guess that either your RabbitMQ SSL configuration is incorrect or that your ssl_options specified are preventing you from connecting.

Also, looking at your output, you appear to be connecting just fine. Any abrupt connection disconnects are happening after your connection is established. These last two lines specify:

DEBUG &#160; &#160; &#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;process &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;231 : Calling &lt;bound method BlockingConnection._on_connection_open of &lt;pika.adapters.blocking_connection.BlockingConnection object at 0x2d6be90&gt;&gt; for &quot;0:Connection.OpenOk&quot;
The connection has been established ok
DEBUG &#160; &#160; &#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;add &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;161 : Added: {'callback': &lt;bound method BlockingConnection._on_connection_closed of &lt;pika.adapters.blocking_connection.BlockingConnection object at 0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments': None, 'calls': 1}

The adapter has registered to be notified when the connection is closed.

If your python app stops there, the connection will be closed abruptly with RabbitMQ since you&#8217;re not telling it to shutdown the connection.

Hope this helps,

Gavin



________________________________
From:&#160;Gavin M. Roy Gavin M. Roy
Reply:&#160;Gavin M. Roy <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">gavinmroy at gmail.com</A>
Date:&#160;January 27, 2014 at 10:56:25 AM
To:&#160;Michael Klishin <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">mklishin at gopivotal.com</A>, Kausik Chattopadhyay <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">connect_kausik at yahoo.com</A>
Subject:&#160;Re: [rabbitmq-discuss] Not able to get ssl working with rabbitMQ server 

Your ssl_options are a pretty important
part of the SSL connection in this code and they are omitted, same
with your other email thread. Without them, it&#8217;s hard to debug for
you. The underlying code is the same regardless of the connection
adapter. If we can see what those are, it should help.

Also, any related snippets from rabbit@[NODE].log could be
helpful.

Gavin


________________________________
 From:&#160;Kausik Chattopadhyay Kausik
Chattopadhyay
Reply:&#160;Kausik Chattopadhyay <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">connect_kausik at yahoo.com</A>, Discussions about RabbitMQ <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbitmq-discuss at lists.rabbitmq.com</A>
Date:&#160;January 27, 2014 at 6:03:05
AM
To:&#160;Michael Klishin <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">mklishin at gopivotal.com</A>
Subject:&#160; Re: [rabbitmq-discuss] Not
able to get ssl working with rabbitMQ server

Hi
Michael,
&gt;<i>So far I
</I>&#160;was working with SelectConnection using pika and it is
good.
&gt;<i>Today I
</I>tried to get a publisher working with BlockingConnection with SSL.
Normal connection works however failing with SSL. The same setup
works with SelectConnection. Any idea ?
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>My
</I>publisher code is as follows:
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>&#160;parameters =
</I>ConnectionParameters(HOST, PORT,
&gt;<i>&#160; &#160; &#160; &#160; &#160;
</I>&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
ssl=True,
&gt;<i>&#160; &#160; &#160; &#160; &#160;
</I>&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
ssl_options=ssl_options)
&gt;<i>connection =
</I>pika.BlockingConnection(parameters)
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>I am
</I>getting the following error:
&gt;<i>
</I>&gt;<i>------------------------------------------
</I>&gt;<i>DEBUG &#160;
</I>&#160; &#160;2014-01-27 16:25:27,901 pika.callback &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160;add &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;161 : Added: {'callback':
&lt;bound method BlockingConnection._on_connection_error of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': False, 'arguments':
None}
&gt;<i>
</I>&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,901 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;add &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160;161 : Added: {'callback':
&lt;bound method BlockingConnection._on_connection_start of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments':
None, 'calls': 1}
&gt;<i>INFO &#160; &#160; &#160;
</I>2014-01-27 16:25:27,901 pika.adapters.base_connection
&#160;_create_and_connect_to_socket &#160; &#160; &#160; &#160;164
:<i> Connecting to 172.16.47.133:5671 with SSL
</I>&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,925 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;process &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160;217 : Processing
0:Connection.Start
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,925 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;_use_one_shot_callback
&#160; &#160; &#160; &#160; &#160; &#160; &#160; 390 : Processing
use of oneshot callback
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,925 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;_use_one_shot_callback
&#160; &#160; &#160; &#160; &#160; &#160; &#160; 392 : 0 registered
uses left
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,925 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;remove &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; 260 : Removing callback #0:
{'callback': &lt;bound method
BlockingConnection._on_connection_start of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments':
None, 'calls': 0}
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;process &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160;231 : Calling &lt;bound method
BlockingConnection._on_connection_start of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt; for &quot;0:Connection.Start&quot;
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;add &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160;161 : Added: {'callback':
&lt;bound method BlockingConnection._on_connection_tune of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments':
None, 'calls': 1}
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;process &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160;217 : Processing 0:Connection.Tune
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;_use_one_shot_callback
&#160; &#160; &#160; &#160; &#160; &#160; &#160; 390 : Processing
use of oneshot callback
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;_use_one_shot_callback
&#160; &#160; &#160; &#160; &#160; &#160; &#160; 392 : 0 registered
uses left
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;remove &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; 260 : Removing callback #0:
{'callback': &lt;bound method
BlockingConnection._on_connection_tune of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments':
None, 'calls': 0}
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;process &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160;231 : Calling &lt;bound method
BlockingConnection._on_connection_tune of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt; for &quot;0:Connection.Tune&quot;
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.connection &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160;_create_heartbeat_checker &#160;
&#160; &#160; &#160; &#160; &#160;946 : Creating a
HeartbeatChecker: 600
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,926 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;add &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160;161 : Added: {'callback':
&lt;bound method BlockingConnection._on_connection_open of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments':
None, 'calls': 1}
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;process &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160;217 : Processing
0:Connection.OpenOk
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;_use_one_shot_callback
&#160; &#160; &#160; &#160; &#160; &#160; &#160; 390 : Processing
use of oneshot callback
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;_use_one_shot_callback
&#160; &#160; &#160; &#160; &#160; &#160; &#160; 392 : 0 registered
uses left
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;remove &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; 260 : Removing callback #0:
{'callback': &lt;bound method
BlockingConnection._on_connection_open of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments':
None, 'calls': 0}
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;process &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160;231 : Calling &lt;bound method
BlockingConnection._on_connection_open of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt; for &quot;0:Connection.OpenOk&quot;
&gt;<i>DEBUG &#160; &#160;
</I>&#160;2014-01-27 16:25:27,927 pika.callback &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160;add &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160; &#160;
&#160; &#160; &#160; &#160; &#160;161 : Added: {'callback':
&lt;bound method BlockingConnection._on_connection_closed of
&lt;pika.adapters.blocking_connection.BlockingConnection object at
0x2d6be90&gt;&gt;, 'only': None, 'one_shot': True, 'arguments':
None, 'calls': 1}
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>---------------------------------------&#160;
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>Serverside log shows:
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>=INFO REPORT====
</I>27-Jan-2014::05:55:47 ===
&gt;<i>accepting AMQP connection
</I>&lt;0.6427.1&gt; (172.16.47.1:56458 -&gt; 172.16.47.133:5671)
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>=WARNING REPORT====
</I>27-Jan-2014::05:55:47 ===
&gt;<i>closing AMQP connection
</I>&lt;0.6427.1&gt; (172.16.47.1:56458 -&gt;
172.16.47.133:5671):
&gt;<i>connection_closed_abruptly
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>On Wednesday, January
</I>22, 2014 8:07 PM, Kausik Chattopadhyay
&lt;<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">connect_kausik at yahoo.com</A>&gt; wrote:
&gt;<i>
</I>&gt;<i>Hi Michael,
</I>&gt;<i>Thanks.
</I>&gt;<i>&#160;Actually the
</I>certificates were owned by root.&#160;
&gt;<i>I tried the
</I>followings:
&gt;<i>1. Adding user
</I>rabbitmq to group root &#160;--- didnt work
&gt;<i>2. Changing the
</I>ownership of all the certficates to rabbitmq:rabbitmq --- didnt
work
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>Finally I tried
</I>this. Created a &quot;rabbitmq&quot; sudoer. Installing rabbitmq server.
Generating the certificates as a &quot;rabbitmq&quot; user --&gt; this
worked
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>Thanks
</I>&gt;<i>Kausik
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>On Wednesday, January
</I>22, 2014 4:54 PM, Michael Klishin &lt;<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">mklishin at gopivotal.com</A>&gt;
wrote:
&gt;<i>
</I>&gt;<i>
</I>&gt;<i>On 22 Jan 2014, at 14:59, Kausik Chattopadhyay &lt;<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">connect_kausik at yahoo.com</A>&gt;
</I>wrote:
&gt;<i>
</I>&gt;&gt;<i> Interestingly, this is the log from <A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbit at master1-sasl.log</A>
</I>&gt;<i>
</I>&gt;<i>There is nothing new in the SASL log.
</I>&gt;<i>
</I>&gt;<i>ecacertfile indicates there was an issue with the CA certificate,
</I>which cannot
&gt;<i>be loaded due to eaccess (according to the other log file). 
</I>&gt;<i>
</I>&gt;<i>--
</I>&gt;<i>MK
</I>&gt;<i>
</I>&gt;<i>Software Engineer, Pivotal/RabbitMQ
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>_______________________________________________
&gt;<i>rabbitmq-discuss mailing list
</I>&gt;<i><A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">rabbitmq-discuss at lists.rabbitmq.com</A>
</I>&gt;<i><A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss</A>
</I>&gt;<i>
</I>-------------- next part --------------
An HTML attachment was scrubbed...
URL: &lt;<A HREF="http://lists.rabbitmq.com/pipermail/rabbitmq-discuss/attachments/20140128/7f70e6b0/attachment.html">http://lists.rabbitmq.com/pipermail/rabbitmq-discuss/attachments/20140128/7f70e6b0/attachment.html</A>&gt;
</PRE>



<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="033358.html">[rabbitmq-discuss] Not able to get ssl working with rabbitMQ server
</A></li>
	<LI>Next message: <A HREF="033254.html">[rabbitmq-discuss] RabbitMQ hangs, does not accept connections
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#33376">[ date ]</a>
              <a href="thread.html#33376">[ thread ]</a>
              <a href="subject.html#33376">[ subject ]</a>
              <a href="author.html#33376">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">More information about the rabbitmq-discuss
mailing list</a><br>
</body></html>
